src/lib/models/emulation/Core.cc

```diff
@@ -204,16 +188,13 @@ void Core::handleException(const std::shared_ptr<Instruction>& instruction) {
 void Core::processExceptionHandler() {
   assert(exceptionHandler_ != nullptr &&
          "Attempted to process an exception handler that wasn't present");
-  if (dataMemory_.hasPendingRequests()) {
-    // Must wait for all memory requests to complete before processing the
-    // exception
-    return;
-  }
 
-  bool success = exceptionHandler_->tick();
-  if (!success) {
-    // Handler needs further ticks to complete
-    return;
+  while (true) {
+    bool success = exceptionHandler_->tick();
+    if (success) {
+      // No more ticks needed to complete exception
+      break;
+    }
   }
```

Comment:
This while loop is pretty horrible. The condition statement (i.e. `true`) doesn't convey the function of the loop. Other options include 

`while (!exceptionHandler_.tick()) {}`
or 
```
bool success = false;
while (!success) {
    success = exceptionHandler_.tick();
}
```

Reply:
Updated